lib/sycamore/sycamore/utils/pytorch_dir.py

```diff
@@ -18,10 +15,16 @@ def get_default_build_root() -> str:
     This directory is **user-specific** so that multiple users on the same
     machine won't meet permission issues.
     """
+    # copied from the torch code.  Importing the file that contains it uses 4GB. This uses 100M
+    import torch._appdirs
+
     return os.path.realpath(torch._appdirs.user_cache_dir(appname="torch_extensions"))
 
 
+@requires_modules("torch", extra="local-inference")
 def get_pytorch_build_directory(name: str, verbose: bool) -> str:
+    import torch.version
```

Comment:
I wonder if we can use this to avoid importing all of torch in that one branch in `choose_device` too?

Reply:
Sounds like a good idea. I'll try it.

Reply:
Actually, looks like that branch isn't executed any more -- we just return cpu on the mac as it seemed to be faster. I was going to change it anyway, but it looks like importing torch.version actually imports torch? (https://github.com/pytorch/pytorch/blob/7600e9b36fd67f09c4bb8d91e6e5a78530c651ac/torch/version.py.tpl#L11). Confusing enough that I'm not going to worry about it at the moment.